CMakeLists.txt

```diff
@@ -95,10 +98,30 @@ zephyr_compile_options(
   -Wformat
   -Wformat-security
   -Wno-format-zero-length
+  -imacros ${AUTOCONF_H}
+)
+
+
+# FIXME
+if(CONFIG_ARCH_POSIX)
+zephyr_compile_options(
+  -Wno-undef
+  -Wno-implicit-function-declaration
```

Comment:
Would it work to move these two into arch/posix/CMakeLists.txt ?

Motivation is same as above.